e2e/e2etests/test_precompiles_staking.go

```diff
@@ -0,0 +1,74 @@
+package e2etests
+
+import (
+	"math/big"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/e2e/runner"
+	"github.com/zeta-chain/node/e2e/utils"
+	"github.com/zeta-chain/node/precompiles/staking"
+)
+
+func TestPrecompilesStaking(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 0, "No arguments expected")
+
+	stakingContract, err := staking.NewIStaking(staking.ContractAddress, r.ZEVMClient)
+	require.NoError(r, err, "Failed to create staking contract caller")
+
+	r.ZEVMAuth.GasLimit = 10000000
```

Comment:
Can we add a defer statement to set gasLimit to previous values at the end of the test? Avoiding interfering with other test